### ISSUE TITLE
Move Release Process to GitHub Actions

### ISSUE THREAD

**Issue by CleanCut** [CleanCut/green#278]
It is time to move the release process from my local machine to GitHub actions.

- Trigger the action manually (via [workflow_dispatch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch))
- Do the equivalent of what `make run` currently does.
  - Test all currently-supported stable versions of CPython, as well as the latest stable pypy3. (`make run` just looks at currently-installed stuff, but that's what I intended to have installed whenever I did a release)
- We should use Actions secrets for the PyPI credentials. PyPI supports GitHub OpenID Connect. I will need to configure the credentials in PyPI and store them in the Actions secrets for the repository.

This is a prerequisite for @sodul to gain publishing privileges.

**Comment by sodul** [comment]
Thanks for granting me maintainer status for green on pipy.org.

I'm looking at https://github.com/marketplace/actions/pypi-publish to handle the releases. The nice thing is that no secret should need to be added which should keep things safe.

I will probably update the 3.5.0 release to use this action.

**Comment by CleanCut** [comment]
Do you need anything from me to enable this?

**Comment by sodul** [comment]
Yes, easiest for you is to grant me ownership on GH and Pypi.org so I can setup Trusted publishing as only owner, not maintainers can do it. You can switch me back to maintainer afterwards.

Alternatively you can follow the instructions to add the GH project as a trusted source, it seems relatively simple but I have never done it:
https://docs.pypi.org/trusted-publishers/adding-a-publisher/

**Comment by CleanCut** [comment]
✅ Done!

![Image](https://github.com/CleanCut/green/assets/5838512/dd9b512c-d780-4978-92ba-f2437ccc5ed2)



![Image](https://github.com/CleanCut/green/assets/5838512/21939a04-7aa2-4e93-b677-80473be48b35)

**Comment by CleanCut** [comment]
I created the corresponding environments.

![Image](https://github.com/CleanCut/green/assets/5838512/8bc9bb35-9c37-486e-877b-8942df4d41a2)

**Comment by sodul** [comment]
Thanks, I was planning on working on the python 3.12.1 compatibility issue this weekend so I can try this at the same time.

**Comment by sodul** [comment]
It worked like charm:
https://github.com/CleanCut/green/releases/tag/4.0.1
https://github.com/CleanCut/green/actions/runs/7919259574
https://pypi.org/project/green/4.0.1/
